lib/view_component/storybook/controls/control_config.rb

```diff
@@ -44,6 +44,18 @@ def csf_value
         def csf_control_params
           { type: type }
         end
+
+        def component_accepts_kwargs?
+          component_params.map(&:first).include?(:keyrest)
+        end
+
+        def component_params
+          @component_params ||= component.instance_method(:initialize).parameters
+        end
+
+        def should_validate_params?
+          !component.nil? && !component_accepts_kwargs?
```

Comment:
Can we avoid the double negative `! nil?` 
```suggestion
          component.present? && !component_accepts_kwargs?
```